hail/src/main/scala/is/hail/expr/ir/Parser.scala

```diff
@@ -0,0 +1,1024 @@
+    }
+  }
+
+  def string_literal(it: TokenIterator): String = {
+    consumeToken(it) match {
+      case x: StringToken => x.value
+      case x: Token => error(x, s"Expected string but found ${ x.getName } '${ x.value }'.")
+    }
+  }
+
+  def string_literals(it: TokenIterator): Array[String] = {
+    punctuation(it, "(")
+    val strings = repUntil(it, string_literal, PunctuationToken(")"))
+    punctuation(it, ")")
+    strings
+  }
+
+  def opt[T](it: TokenIterator, f: (TokenIterator) => T)(implicit tct: ClassTag[T]): Option[T] = {
+    it.head match {
+      case x: IdentifierToken if x.value == "None" =>
+        consumeToken(it)
+        None
+      case _ =>
+        Some(f(it))
+    }
+  }
+
+  def repsepUntil[T](it: TokenIterator,
+    f: (TokenIterator) => T,
+    sep: Token,
+    end: Token)(implicit tct: ClassTag[T]): Array[T] = {
+    val xs = new ArrayBuilder[T]()
+    while (it.hasNext && it.head != end) {
+      xs += f(it)
+      if (it.head == sep)
+        consumeToken(it)
+    }
+    xs.result()
+  }
+
+  def repUntil[T](it: TokenIterator,
+    f: (TokenIterator) => T,
+    end: Token)(implicit tct: ClassTag[T]): Array[T] = {
+    val xs = new ArrayBuilder[T]()
+    while (it.hasNext && it.head != end) {
+      xs += f(it)
+    }
+    xs.result()
+  }
+
+  def decorator(it: TokenIterator): (String, String) = {
+    punctuation(it, "@")
+    val name = identifier(it)
+    punctuation(it, "=")
+    val desc = string_literal(it)
+    (name, desc)
+  }
+
+  def type_field(it: TokenIterator): (String, Type) = {
+    val name = identifier(it)
+    punctuation(it, ":")
+    val typ = type_expr(it)
+    while (it.hasNext && it.head == PunctuationToken("@")) {
+      decorator(it)
+    }
+    (name, typ)
+  }
+
+  def type_exprs(it: TokenIterator): Array[Type] = {
+    punctuation(it, "(")
+    val types = repUntil(it, type_expr, PunctuationToken(")"))
+    punctuation(it, ")")
+    types
+  }
+
+  def type_expr(it: TokenIterator): Type = {
+    val req = it.head match {
+      case x: PunctuationToken if x.value == "+" =>
+        consumeToken(it)
+        true
+      case _ => false
+    }
+
+    val typ = identifier(it) match {
+      case "Interval" =>
+        punctuation(it, "[")
+        val pointType = type_expr(it)
+        punctuation(it, "]")
+        TInterval(pointType)
+      case "Boolean" => TBoolean()
+      case "Int32" => TInt32()
+      case "Int64" => TInt64()
+      case "Int" => TInt32()
+      case "Float32" => TFloat32()
+      case "Float64" => TFloat64()
+      case "String" => TString()
+      case "Locus" =>
+        punctuation(it, "(")
+        val id = identifier(it)
+        punctuation(it, ")")
+        ReferenceGenome.getReference(id).locusType
+      case "LocusAlleles" =>
+        punctuation(it, "(")
+        val id = identifier(it)
+        punctuation(it, ")")
+        ReferenceGenome.getReference(id).intervalType
```

Comment:
I think it's used for backwards compatibility with older matrix tables and tables. I'm not sure what the correct type should be. Clearly it was a bug before. Shouldn't have been `.locusType`, but I thought it was supposed to be `TInterval(TLocus)` which is what it is now in my code. But the name `LocusAlleles` would be wrong in that case. There is a `LocusAlleles` function that parses the variant in the function registry. I think I originally meant it to be `LocusInterval`. Not sure what to do -- just delete it since clearly it's wrong?

Reply:
Ok. The original thing was `Variant(rg) -> TVariant(rg)`. I'll make it have the correct type.

Reply:
I looked at the files we have for backwards compatibility and `LocusAlleles` is never a type there. I think we can go ahead and delete it. `src/test/resources/backward_compatability/1.0.0/`